packages/@lwc/ssr-compiler/src/compile-template/shared.ts

```diff
@@ -10,14 +10,17 @@ import { reservedKeywords } from '@lwc/shared';
 import { Node as IrNode } from '@lwc/template-compiler';
 import { esTemplate } from '../estemplate';
 
+import { TransformerContext } from './types';
 import type {
     ImportDeclaration as EsImportDeclaration,
     Statement as EsStatement,
     Expression as EsExpression,
+    MemberExpression as EsMemberExpression,
+    Identifier as EsIdentifier,
 } from 'estree';
 
 export const bImportHtmlEscape = esTemplate`
-    import { htmlEscape } from '@lwc/shared';
+    import { htmlEscape } from '@lwc/ssr-runtime';
```

Comment:
It's simpler just to have everything at runtime imported from ssr-runtime, rather than introducing more dependencies at runtime.